### ISSUE TITLE
Prettierrc support/default usage

### ISSUE THREAD

**Issue by djvs** [prettier/vim-prettier#137]
**Do you want to request a *feature* or report a *bug*?**

Kind of both?

**What is the current/expected behavior?**

Expect vim-prettier to somehow respect .prettierrc settings

**What version of `vim-prettier` are you using - (output of `:PrettierVersion`) ?**

0.2.6

**What version of `prettier` are you using - (output of  `:PrettierCliVersion`) ?**

1.12.1

**What is your `prettier` executable path - (output of  `:PrettierCliPath`) ?**

~/.vim/plugged/vim-prettier/node_mmodules/.bin/prettier

**Did this work in previous versions of vim-prettier and/or prettier ?**

I doubt it

---------

Basically, I just want vim-prettier to respect a standard .prettierrc/.prettierrc.js/.prettierrc.yml/.prettierrc.json file that's somewhere above the currently edited file in the filesystem.  I'm not sure why the configuration options are meant to be different.  

This is also not working in .vimrc (which is the main thing I'm trying to do):

let g:prettier#config#semi = 'false'

**Comment by mitermayer** [comment]
Hi @djvs

The behavior you have described is supported by vim prettier by default. 

I am more than happy to try debugging with you why it may seem to not have worked on your case.

Could you please check that the project you are having problems with does not include some sort of prettier configuration on its package.json or any other related prettier configuration file ?

**Comment by djvs** [comment]
It has one, which I added while trying to get rid of semicolons this morning:

.prettierrc.yml
`semi: false`

There's also the same file one directory up.

**Comment by mitermayer** [comment]
Hi @djvs ,

I haven't be able to replicate your issue. Would be great if you can provide a bit more information to help us see if we can find a solution for your problem.

* What is the OS your are using ?
* Does the path to the file you are editing has any weird characters or white spaces ?
* Are you sure there is no other configuration file being declared on the project ?
* How did you install `vim-prettier`, did you do it from the `master` branch ? 
* What is your `.vimrc` vim-prettier configuration ?

thanks for your patience, hopefuly we can help figure out why you are having problems

**Comment by mitermayer** [comment]
I will close this issue for now as I can't replicate it, if @djvs  can provide a bit more info in order to aid reproducing or any other community member start experiencing this issue as well please feel free to reopen this issue

**Comment by vaclavbohac** [comment]
@mitermayer The same thing happens to me:

* What is the OS your are using ?
    * Mac Os

* Does the path to the file you are editing has any weird characters or white spaces ?
    * `/Users/vaclavbohac/repo/demagog/demagog/app/javascript/packs/admin.ts`

* Are you sure there is no other configuration file being declared on the project?
    * Yes

* How did you install vim-prettier, did you do it from the master branch ?
    * Via pathogen `git clone https://github.com/prettier/vim-prettier ~/.vim/bundle`

* What is your .vimrc vim-prettier configuration?
    * None

Other than that:
* vim-prettier is in latest master branch
* It's TypeScript project
* `.prettierrc.json`
```json 
{
  "printWidth": 100,
  "parser": "typescript",
  "semi": true,
  "singleQuote": true,
  "trailingComma": "all",
  "arrowParens": "always"
}
```

I can setup minimalistic repo replicating the issue if that would be helpful.

**Comment by mitermayer** [comment]
Hi @vaclavbohac,

Would be great if you could setup a minimal way to reproduce this, and we would be able to look into it. Can you also provide the outputs of 

from within vim:

* `:PrettierCliVersion` 
* `:PrettierCliPath`
* `:PrettierVersion`

and also the output of `vim --version` ?

thanks a lot!

**Comment by mitermayer** [comment]
I have tried to replicate this by creating a dummy file:

```typescript
function a() {
  return "foo"
}
```

and creating a config file `.prettierrc.json` with the contents

```json
{
  "parser": "typescript",
  "printWidth": 100,
  "semi": true,
  "singleQuote": true,
  "trailingComma": "all",
  "arrowParens": "always"
}
```

Opening the file and running `:Prettier` and `:PrettierAsync` and both times i get the same output:

```typescript
function a() {
  return 'foo';
}
```

Not entirely sure on how to replicate this issue, @docwhat, @SamHowie  do you guys think you could try replicating this on your machines to see if that is OS specific since im on linux ?

**Comment by docwhat** [comment]
This is fixed in the `release/1.x` branch; specifically commit 3e76ead2077ec866761bf243dd8750f4dc69f945

The option `g:prettier#config#config_precedence` in 1.x defaults to `prefer-file` instead of 0.x's default of 'cli-override`.  With `cli-override` then `.prettierrc` files are ignored.

@djvs Can you take the `release/1.x` branch to see if it fixes your problem?  See the readme for the new flags you can play with, tho the defaults should be decent; especially if you rely on `.prettierrc` files.

**Comment by vaclavbohac** [comment]
@mitermayer 

* `:PrettierCliVersion` 1.2.1
* `:PrettierCliPath` `~/my-project/node_modules/.bin/prettier`
* `:PrettierVersion` 0.2.7
* `vim --version` 8.0.1800

**Comment by mitermayer** [comment]
Hi @vaclavbohac,

You are using a very old prettier cli 1.2.1 installation and at that stage prettier did not support any custom configuration
